src/sage/symbolic/expression_conversions.py

```diff
@@ -1245,6 +1245,16 @@ def composition(self, ex, operator):
             sage: a.composition(sin(pi/7), sin)
             0.4338837391175581? + 0.?e-18*I
 
+            sage: x = SR.var('x')
+            sage: a.composition(complex_root_of(x^3 - x^2 - x - 1, 0), complex_root_of)
+            1.839286755214161?
+            sage: a.composition(complex_root_of(x^5 - 1, 3), complex_root_of)
+            0.3090169943749474? - 0.9510565162951536?*I
+            sage: a.composition(complex_root_of(x^2 + 1, 0), complex_root_of)
+            1.?e-684 - 0.9999999999999999?*I
+            sage: a.composition(complex_root_of(x^2 + 1, 1), complex_root_of)
```

Comment:
why this duplicated doctest ?

Reply:
These are two different lines. In the first one index is `0` (corresponding to `-I`) and the second one is `1` (corresponding to `+I`).

Reply:
oops, désolé